packages/components/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,155 @@
+import 'react-native'
+import { fireEvent, render, screen } from '@testing-library/react-native'
+import React from 'react'
+
+import { Checkbox } from './Checkbox'
+import { Icon } from '../Icon/Icon'
+
+const mockedColorScheme = jest.fn()
+
+jest.mock('react-native/Libraries/Utilities/useColorScheme', () => {
+  return {
+    default: mockedColorScheme,
+  }
+})
+
+describe('Checkbox', () => {
+  const onPressSpy = jest.fn()
+
+  const commonProps = {
+    header: 'Header text',
+    description: 'Description text',
+    hint: 'Hint text',
+    label: 'Label text',
+    testID: 'testCheckbox',
+    onPress: onPressSpy,
+  }
+
+  const errorMsg = 'Error text'
+
+  const getIconName = async () => {
+    const icon = await screen.root.findByType(Icon)
+    return icon.props.name
+  }
+
+  describe('Basic tests', () => {
+    it('initializes correctly', () => {
+      render(<Checkbox {...commonProps} />)
+      expect(screen.getByTestId('testCheckbox')).toBeTruthy()
+    })
+
+    it('renders header, description, hint, icon, and label text', async () => {
+      render(<Checkbox {...commonProps} />)
+      expect(screen.getByText('Header text')).toBeOnTheScreen()
+      expect(screen.getByText('Description text')).toBeOnTheScreen()
+      expect(screen.getByText('Label text')).toBeOnTheScreen()
+      expect(screen.getByText('Hint text')).toBeOnTheScreen()
+      expect(screen.queryByText('(*Required)')).not.toBeOnTheScreen()
+      expect(await getIconName()).toBe('CheckBoxOutlineBlank')
+    })
+
+    it('renders required indicator', async () => {
```

Comment:
Wonder if this test should be more specific to validate against it being on label since the behavior is different for CheckboxGroup and in this manner it would pass just the same.